wingtips-lightstep/src/main/java/com/nike/wingtips/lightstep/WingtipsToLightStepLifecycleListener.java

```diff
@@ -0,0 +1,152 @@
+
+import io.opentracing.Tracer;
+import com.lightstep.tracer.shared.SpanContext;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicLong;
+
+/**
+ * A{@link SpanLifecycleListener} that converts Wingtips {@link Span}s to [LightStep implementation of] an OpenTracing
+ * {@link io.opentracing.Span}, and sends that data to the LightStep Satellites via
+ * the {@link com.lightstep.tracer.jre.JRETracer}.
+ *
+ * We're adapting some of the prior work built in the Wingtips Zipkin2 plugin to handle conversion of span/trace/parent
+ * IDs as well as frequency gates for exception logging.
+ *
+ * Required options used in the constructor are the LightStep access token (generated in project settings within
+ * LightStep), service name (which will be assigned to all spans), Satellite URL and Satellite Port, which should both
+ * reflect the address for the load balancer in front of the LightStep Satellites.
+ *
+ * @author parker@lightstep.com
+ */
+
+public class WingtipsToLightStepLifecycleListener implements SpanLifecycleListener {
+
+    // we borrowed the logging and exception log rate limiting from the Zipkin plugin.
+    private final Logger lightStepToWingtipsLogger = LoggerFactory.getLogger("LIGHTSTEP_SPAN_CONVERSION_OR_HANDLING_ERROR");
+
+    private final AtomicLong spanHandlingErrorCounter = new AtomicLong(0);
+    private long lastSpanHandlingErrorLogTimeEpochMillis = 0;
+    private static final long MIN_SPAN_HANDLING_ERROR_LOG_INTERVAL_MILLIS = TimeUnit.SECONDS.toMillis(60);
+
+    protected String serviceName;
+    protected String accessToken;
+    protected String satelliteUrl;
+    protected int satellitePort;
+
+    protected Tracer tracer = null;
+
+    // Basic constructor which requires values to configure tracer and point span traffic from the transport library
+    // to the LightStep Satellites.
+    public WingtipsToLightStepLifecycleListener(String serviceName, String accessToken, String satelliteUrl, int satellitePort) {
+        this.serviceName = serviceName;
+        this.accessToken = accessToken;
+        this.satelliteUrl = satelliteUrl;
+        this.satellitePort = satellitePort;
+
+        try {
+            this.tracer = new com.lightstep.tracer.jre.JRETracer(
+                    new com.lightstep.tracer.shared.Options.OptionsBuilder()
+                            .withAccessToken(this.accessToken)
+                            .withComponentName(this.serviceName)
+                            .withCollectorHost(this.satelliteUrl)
+                            .withCollectorPort(this.satellitePort)
+                            .withVerbosity(4)
+                            .build()
+            );
+        } catch (Throwable ex) {
+            lightStepToWingtipsLogger.warn(
+                    "There has been an issue with initializing the LightStep tracer: ", ex.toString());
```

Comment:
Go ahead and let the full stack trace show up in the logs here (i.e. get rid of the `.toString()` on the exception).